lib/Dialect/Calyx/Export/CalyxEmitter.cpp

```diff
@@ -1005,18 +1009,25 @@ void Emitter::emitLibraryPrimTypedByFirstOutputPort(
 
 void Emitter::emitLibraryFloatingPoint(Operation *op) {
   auto cell = cast<CellInterface>(op);
-  // magic number for the index of `left/right` input port for
-  // `AddF`/`MulF`/`CompareF`; `in` input port for `FpToInt`.
-  auto inputPorts = cell.getInputPorts();
-  assert(inputPorts.size() >= 2 && "There should be at least two input ports");
-  size_t inputPortIndex = inputPorts.size() - 2;
-  unsigned bitWidth =
-      cell.getInputPorts()[inputPortIndex].getType().getIntOrFloatBitWidth();
+  unsigned fpBitWidth;
+  if (auto intToFpOp = dyn_cast<IntToFpOpIEEE754>(op))
+    // `std_intToFp` has the float point value in the first output port.
+    fpBitWidth = cell.getOutputPorts()[0].getType().getIntOrFloatBitWidth();
+  else {
+    auto inputPorts = cell.getInputPorts();
+    assert(inputPorts.size() >= 2 &&
+           "There should be at least two input ports");
+    // magic number for the index of `left/right` input port for
+    // `AddF`/`MulF`/`CompareF`; `in` input port for `FpToInt`.
+    size_t inputPortIndex = inputPorts.size() - 2;
+    fpBitWidth =
+        cell.getInputPorts()[inputPortIndex].getType().getIntOrFloatBitWidth();
+  }
```

Comment:
Nit: let's place this in a function. It is getting quite complex now.

```
unsigned getFPBitWidth(CellInterface cell);
```

Then it can be used below as well, if I'm not mistaken.

Reply:
totally, just refactored!